### DISCUSSION TITLE
Position-Smoothing

### DISCUSSION THREAD

**Discussion by c-mahn** [organicmaps/organicmaps#1554] [Ideas]
Hello,
I'd like to know, if it's possible to have the Position-Arrow, that indicates my position, to smooth the location updates. When navigating the Arrow doesn't jump, but when I'm not navigating it does. I would love to have an option, where it would smooth out the location.

**Comment by biodranik** [comment]
There should be already an issue for that. What is your use-case? Why Navigation mode is not suitable for you?

**Reply by c-mahn** [reply]
Sometimes I just want to see the surroundings pass by when I'm a passenger in a car or train. Other times I'm just walking down a street with the map open, in order to see the shops in neighboring streets. In situations like these it would be nice, if the map wouldn't make a stop-and-go motion and rather have one fluid motion.

**Reply by biodranik** [reply]
It's hard to implement for a foot walk because your position should be predicted and some artificial animation should be played. That works best with cars and trains, they do not change their direction often.

**Reply by c-mahn** [reply]
You could make the behavior to only appear when moving faster than 5 km/h. So it will be accurate when walking slow or standing around, but will activate on faster movement.

**Reply by c-mahn** [reply]
I don't know, how the app works behind the scenes, but it I'm thinking it would be like a spline motion or something similar, with one point of the last position, one point of the linear predicted position and one point with the new coordinate.

**Reply by Rydyn** [reply]
The increasing  prevalence of Android Radios adds another use case.  It's nice to have a map and nearby shops & services shown on your dashboard display while driving (and not navigating).  The jumpy screen updates make you carsick even out of the corner of your eye.  Also there isn't a concern for battery drain with an installed radio.

**Reply by Rydyn** [reply]
Usage I mentioned is related to #1058 and #1155

**Comment by biodranik** [comment]
See #1459

**Reply by c-mahn** [reply]
Oh, yeah. This is exactly what I was thinking of.
